### ISSUE TITLE
Lineage does not seem to be saving the elapsed time

### ISSUE THREAD

**Issue by jfischer** [data-workspaces/data-workspaces-core#20]

**Comment by jfischer** [comment]
Utkarsh saw this as well with his script step.

**Comment by musically-ut** [comment]
This is how I'm using the API:

```python
def run(...):

    builder = (
        LineageBuilder()
        .as_script_step()
        .with_parameters({
            'force': force,
            'min_count': min_count
        })
        .as_results_step(os.path.join(results_dir, 'preprocessing/'))
    )

    builder = builder.with_input_paths(data_files)

    # ...

    with builder.eval() as lineage:
        # ....
        lineage.add_output_path(output_csv)
        lineage.write_results({
            'size_filtered': len(idx_filtered),
            'size_raw': df_grouped.shape[0],
        })
```

The output from this is the following:

```bash
(ssh session)(tmux)$ time <script-command>
[2019-07-05T13:42:43] Total files found =  101
[2019-07-05T13:42:43] Starting reading ...
[...]
[2019-07-05T13:43:30] Keeping 2099542 rows ...
Wrote results to results:results.json
[2019-07-05T13:44:38] Done.

real    1m59.279s
user    8m1.814s
sys     0m21.691s


(ssh session)(tmux)$ cat results/results.json
{
  "step": "swift_to_hlr",
  "start_time": "2019-07-05T13:42:43.082094",
  "execution_time_seconds": null,
  "parameters": {
    "force": false,
    "min_count": 1
  },
  "run_description": null,
  "metrics": {
    "size_filtered": 966576,
    "size_raw": 966576
  }
}
```

Hence, `"execution_time_seconds"` is still `null`.

**Comment by musically-ut** [comment]
I'm using the following version of `dws`:

```python
In [1]: import dataworkspaces as dws
In [3]: dws.__version__
Out[3]: '1.0.4'

**Comment by jfischer** [comment]
The elapsed time was being set in the exit of the lineage object's context manager. write_results() is typically called before that. Fixed by setting it in either write_results() or complete(), whichever is called first.

Also, corrected some inaccuracies in the docs and wrote a new test case for write_results().

The changes are available on PyPi in version 1.0.6.
